euclidlib/photo/_phz.py

```diff
@@ -2,43 +2,155 @@
 
 from collections.abc import Mapping
 from os import PathLike
-from typing import Any
+from typing import TYPE_CHECKING
 
 import fitsio  # type: ignore [import-not-found]
 import numpy as np
-from numpy.typing import NDArray
+
+if TYPE_CHECKING:
+    from typing import Any
+    from numpy.typing import ArrayLike, NDArray
+
+
+def _hist2dist(x: NDArray[Any], y: NDArray[Any]) -> NDArray[Any]:
+    """
+    Convert histogram to distribution.
+    """
+    *dims, n = y.shape
+    z = np.zeros((*dims, n + 1), dtype=y.dtype.base)
+    np.cumsum(y, axis=-1, out=z[..., 1:])
+    result: NDArray[Any] = np.gradient(z, x, axis=-1, edge_order=1)
+    return result
+
+
+def write_nz(
```

Comment:
Although I agree with keeping `write_nz` private for the moment, why aren't we consistent already with names?

Let's use `read_redshift_distributions` and `write_redshift_distributions`...?

Reply:
I don't like the "read" prefix because most or all public functions should be for reading. The names are very long already.

I used "write_nz()" precisely because it's not a public function (at the moment).

Reply:
I think we should iterate a bit on this, as making decisions for names will pave the way of the next implementations. If we do not include 'read/reading' for any function, we are just giving for granted that `euclidlib` will only `read` in the future. In that case, I do not see an easy way to include `writing` any time soon as we will need to specify in the name somehow about what they do, and thus, we will be inconsistent between names for 'reading' and names for 'writing'.

Reply:
I have implemented this as a test, so we can see how it looks. We can revert if you don't like it.

Reply:
I love the solution! Elegant and user-friendly. If you are happy with it, we are ready to go.